Modify logging to reduce build size further

soylentOrange opened the pull request:
- Modified the logging code in main.cpp to exclude it thoroughly from the build (and saving 12 kbytes) when logger support is disabled (i.e. `; -D MYCILA_LOGGER_SUPPORT` is commented out in `platformio.ini`).
- Added test results with different options to the README.md
- Also, I took the liberty to add my favorite boards

mathieucarbou commented:
Thanks! I will have a look. There are many things in your pr and I see where you want to go.
Although we can change the features included in safeboot, the partition size has to stay the same. It is a good idea to activate features per board.

mathieucarbou commented:
@soylentOrange : could you please tell me why you had to redefine the existing board `lolin_s2_mini` ? Did you try generate and use a safeboot image with `board = lolin_s2_mini` ? 
Same question for the `waveshare_esp32_s3_tiny`, which is an esp32s3...

mathieucarbou commented:
@soylentOrange : I am not able to push a commit, probably because this is your main branch.

So I have opened PR #11 with a squash of all commits plus some cleanup I did.

I will merge it, but please do not hesitate to rebase your PR on top of main after if you feel some things still need adjustments.

Thanks a lot for the updates! 
I did not think about leaving off arduino logging with still activate mycila logger - which brings down flash size a lot and allow for mDNS to be included. Really nice!

soylentOrange commented:
> @soylentOrange : could you please tell me why you had to redefine the existing board `lolin_s2_mini` ? Did you try generate and use a safeboot image with `board = lolin_s2_mini` ? Same question for the `waveshare_esp32_s3_tiny`, which is an esp32s3... For the safeboot image, it often does not variant which variant it is - it will in most case work with the inherited MCU.

@mathieucarbou Thanks for merging the code into your repository. I really like this library, yet I have made some additions to it that I wanted to share, when they are helpful.
Regarding the two board that I added: I used a local safeboot-folder in one of my platformio projects that I built with lolin_s2_mini and waveshare_esp32_s3_tiny. As the waveshare_esp32_s3_tiny wasn't (still isn't?) an officially supported board, I added it to the project and had to include it in SafeBoot as well. Probably should have just removed the boards before submitting my PR as it's only relevant in the special case of including it via `custom_safeboot_dir`, otherwise I could have just used one of the precompiled builds. Actually, I cannot remember the reason for duplicating lolin_s2_mini.json, it's identical to the one provided by platformio...
